src/linear_algebra/orientation.lean

```diff
@@ -526,6 +579,49 @@ begin
     simp [h₁, h₂]
 end
 
+/-- If the index type has cardinality equal to the finite dimension, an orientation equals the
+negation of another orientation if and only if they are not equal. -/
+lemma ne_iff_eq_neg (x₁ x₂ : orientation R M ι) (h : fintype.card ι = finrank R M) :
+  x₁ ≠ x₂ ↔ x₁ = -x₂ :=
+⟨λ hn, (eq_or_eq_neg x₁ x₂ h).resolve_left hn, λ he, he.symm ▸ (module.ray.ne_neg_self x₂).symm⟩
+
 end orientation
 
+namespace alternating_map
+
+variables [fintype ι] [finite_dimensional R M]
+
+open finite_dimensional
+
+/-- If the index type has cardinality equal to the finite dimension, composing an alternating
+map with the same linear equiv on each argument gives the same orientation if and only if the
+determinant is positive. -/
+lemma orientation_comp_linear_equiv_eq_iff_det_pos (f : alternating_map R M R ι) (hf : f ≠ 0)
+  (g : M ≃ₗ[R] M) (h : fintype.card ι = finrank R M) :
+  ray_of_ne_zero R (f.comp_linear_map ↑g)
+                   ((not_iff_not.2 (f.comp_linear_equiv_eq_zero_iff g)).2 hf)
```

Comment:
The LHS here is defeq to `(ray_of_ne_zero R f hf).map g` I think

Reply:
At which point both sides only refer to `f` via `ray_of_ne_zero R f hf`, so you can generalize to `f : orientation R M ι`